RadioControl_rf69/RadioControl_rf69.ino

```diff
@@ -0,0 +1,175 @@
+    driver.setEncryptionKey(key);
+  }
+  if (DEBUG) {
+    SerialUSB.println("Setup Complete: ");
+    if  (TRANSMITTER) {
+      SerialUSB.println("transmitter mode");
+    }
+    else {
+      SerialUSB.println("receiver mode");
+    }
+  }
+}
+
+void loop() {
+#if TRANSMITTER
+  txloop();
+#else
+  rxloop();
+#endif
+}
+
+void readInputs() {
+  txData.turn = analogRead(TURN_PIN);
+  txData.throttle = analogRead(THROTTLE_PIN);
+  txData.autonomous = digitalRead(AUTO_PIN);
+  txData.ebrake = digitalRead(EBRAKE_PIN);
+}
+
+static bool received = false;
+void setLights() {
+  digitalWrite(TX_LED_LINK, received);
+}
+
+void txloop()
+{
+  readInputs();
+  // Send a message to receiver unit
+  driver.send((unsigned char*)(&txData), lenTx);
+  driver.waitPacketSent();
+  timeSent = millis();
+  // Now wait for a reply
+  if (driver.waitAvailableTimeout(BACKOFF)) {
+    // we have a reply
+    if (driver.recv((unsigned char*)(&rxData), &lenRx)) {
+      // payload is the right size
+      received = true;
+      txData.rssi = driver.rssiRead();
+      setLights();
+      timeRecv = millis();
+      if (DEBUG)
+        SerialUSB.println(String(rxData.rssi / 2.0));
+
+      unsigned long interval = timeRecv - timeSent;
+      unsigned long msDelay = BACKOFF - interval;
+      if (msDelay < BACKOFF) {
+        delay(msDelay);
+      }
+      else {
+        delay(BACKOFF);
+      }
+    }
+
+    else {
+      // payload is not the right size
+      received = false;
+      setLights();
+      if (DEBUG)
+        SerialUSB.println("!");
+    }
+  }
+  else {
+    // we did not get a reply
+    received = false;
+    setLights();
+    if (DEBUG)
+      SerialUSB.println("?");
+  }
+}
+
+void rxloop() {
+  if (driver.available()) {
+    // a packet is availible
+    if (driver.recv((unsigned char*)(&txData), &lenTx)) {
+      // payload is the right size
+      if (timeRecv > 0)
+        timeSent = timeRecv;
+      timeRecv = millis();
+      rxData.rssi = driver.rssiRead();
+      driver.send((unsigned char*)(&rxData), lenRx); // packet sent to transmitter
+      driver.waitPacketSent();
+      if (DEBUG) {
+        printTx(false);
+        unsigned long duration = timeRecv - timeSent;
+        SerialUSB.println(","
+                          + String(duration));
+      }
+    }
+    else {
+      // payload was the wrong size
+      if (DEBUG)
+        SerialUSB.println("!");
+    }
+  }
+}
+
+void printTx(bool newline) {
+  // output formatted as CSV for ease of review/logging
+  SerialUSB.print(String((rxData.rssi / 2.0))
+                  + ","
+                  + String(txData.throttle)
+                  + ","
+                  + String(txData.turn)
+                  + ",");
+
+  if (txData.ebrake) {
+    SerialUSB.print("E,");
+  }
+  else {
+    SerialUSB.print(",");
+  }
+
+  if (txData.autonomous) {
+    SerialUSB.print("A,");
+  }
+  else {
+    SerialUSB.print(",");
+  }
+  if (newline)
+    SerialUSB.print("\n");
```

Comment:
Just FYI, if you use println() that will use whatever the system's newline is.  Also, if you could have prints in interrupt routines, then those can sneak in between your separate print(...) calls.  You could avoid that by forming the whole string first, and then doing one print(...).  E.g. put "E" or "" in a variable, and "A" or "" in another, in your if statements, then line up the whole schmeer as a string expression.